warehouse/query-core/src/main/java/datawave/query/postprocessing/tf/FieldValue.java

```diff
@@ -0,0 +1,131 @@
+package datawave.query.postprocessing.tf;
+
+import datawave.util.StringUtils;
+import org.apache.accumulo.core.data.Key;
+import org.apache.hadoop.io.Text;
+
+/**
+ * A field name and value which is sorted on {@code <value>\0<name>}
+ */
+@Deprecated
+public class FieldValue implements Comparable<FieldValue> {
+    private int nullOffset;
+    private String valueField;
+    
+    public FieldValue(String field, String value) {
+        this.nullOffset = value.length();
+        this.valueField = value + '\0' + field;
+    }
+    
+    /**
+     * A distance between this field value and another. Here we want a distance that correlates with the number of keys between here and there for the same.
+     * Essentially we want the inverse of the number of bytes that match. document.
+     *
+     * @param fv
+     * @return a distance between here and there (negative means there is before here)
+     */
+    public double distance(FieldValue fv) {
+        byte[] s1 = getValueField().getBytes();
+        byte[] s2 = fv.getValueField().getBytes();
+        int len = Math.min(s1.length, s2.length);
+        
+        int matches = 0;
+        int lastCharDiff = 0;
+        
+        for (int i = 0; i <= len; i++) {
+            lastCharDiff = getValue(s2, i) - getValue(s1, i);
+            if (lastCharDiff == 0) {
+                matches++;
+            } else {
+                break;
+            }
+        }
+        
+        return Math.copySign(1.0d / (matches + 1), lastCharDiff);
+    }
+    
+    private int getValue(byte[] bytes, int index) {
+        if (index >= bytes.length) {
+            return 0;
+        } else {
+            return bytes[index];
+        }
+    }
+    
+    public String getValueField() {
+        return valueField;
+    }
+    
+    public String getField() {
+        return valueField.substring(nullOffset + 1);
+    }
+    
+    public String getValue() {
+        return valueField.substring(0, nullOffset);
+    }
+    
+    @Override
+    public int compareTo(FieldValue o) {
+        return valueField.compareTo(o.valueField);
+    }
+    
+    @Override
+    public int hashCode() {
+        return valueField.hashCode();
+    }
+    
+    @Override
+    public boolean equals(Object obj) {
+        if (obj instanceof FieldValue) {
+            return valueField.equals(((FieldValue) obj).valueField);
+        }
+        return false;
+    }
+    
+    @Override
+    public String toString() {
+        return getField() + " -> " + getValue();
+    }
+    
+    public static FieldValue getFieldValue(Key key) {
+        return getFieldValue(key.getColumnQualifier());
+    }
+    
+    public static FieldValue getFieldValue(Text cqText) {
+        if (cqText == null) {
+            return null;
+        }
+        return getFieldValue(cqText.toString());
+    }
+    
+    public static FieldValue getFieldValue(String cq) {
```

Comment:
Consider returning `Optional<FieldValue>` for these static methods instead. This will imply that it's possible for a null value to be returned, and reduce the chance of NPEs. It will also help in any functional programming usages.